Kernel/Locking/Mutex.h

```diff
@@ -33,23 +34,23 @@ class Mutex {
     ~Mutex() = default;
 
     void lock(Mode mode = Mode::Exclusive, LockLocation const& location = LockLocation::current());
-    void restore_lock(Mode, u32, LockLocation const& location = LockLocation::current());
+    void restore_exclusive_lock(u32, LockLocation const& location = LockLocation::current());
 
     void unlock();
-    [[nodiscard]] Mode force_unlock_if_locked(u32&);
+    [[nodiscard]] Mode force_unlock_exclusive_if_locked(u32&);
     [[nodiscard]] bool is_locked() const
     {
         SpinlockLocker lock(m_lock);
         return m_mode != Mode::Unlocked;
     }
-    [[nodiscard]] bool is_locked_by_current_thread() const
+
+    [[nodiscard]] bool is_exclusively_locked_by_current_thread() const
     {
         SpinlockLocker lock(m_lock);
-        if (m_mode == Mode::Exclusive)
-            return m_holder == Thread::current();
-        if (m_mode == Mode::Shared)
-            return m_shared_holders.contains(Thread::current());
-        return false;
+        VERIFY(m_mode != Mode::Shared);
```

Comment:
That's a bit aggressive for an `is_...` function, if this is being used as a "you really shouldn't be asking this question if you've got a shared lock", perhaps it would be nicer to have it say that before blowing up?

Reply:
Hmm, that was the point, yes. I could just add a dbgln, but a dbgln in a `is_...` function is a bit heavy for my taste. I could also just make it return false, since if it's shared it's clearly not exclusively locked by the current thread.

Reply:
(do you have a preference?)

Reply:
(I just meant a comment)

Reply:
Oh sure, a comment would work, will do :sweat_smile: